src/test/java/ch/jalu/typeresolver/classutil/ClassUtilTest.java

```diff
@@ -0,0 +1,442 @@
+            String result = ClassUtil.processClassByType(clazz, typeCallback);
+
+            // then
+            assertThat(result, equalTo("enum[" + TimeUnit.class.getName() + "]"));
+        }
+
+        @Test
+        void shouldUseEnumEntryCallback() {
+            // given
+            ClassTypeCallback<String> typeCallback = new CallbackTestImpl();
+            Class<?> clazz = EnumSample.M.getClass();
+
+            // when
+            String result = ClassUtil.processClassByType(clazz, typeCallback);
+
+            // then
+            assertThat(result, equalTo("enumEntry[" + EnumSample.class.getName() + "]"));
+        }
+
+        @Test
+        void shouldUseAnnotationCallback() {
+            // given
+            ClassTypeCallback<String> typeCallback = new CallbackTestImpl();
+            Class<?> clazz = Override.class;
+
+            // when
+            String result = ClassUtil.processClassByType(clazz, typeCallback);
+
+            // then
+            assertThat(result, equalTo("annotation[" + Override.class.getName() + "]"));
+        }
+
+        @Test
+        void shouldUsePrimitiveTypeCallback() {
+            // given
+            ClassTypeCallback<String> typeCallback = new CallbackTestImpl();
+            Class<?> clazz = int.class;
+
+            // when
+            String result = ClassUtil.processClassByType(clazz, typeCallback);
+
+            // then
+            assertThat(result, equalTo("primitiveType[" + int.class.getName() + "]"));
+        }
+
+        @Test
+        void shouldUseArrayTypeCallback() {
+            // given
+            ClassTypeCallback<String> typeCallback = new CallbackTestImpl();
+            Class<?> clazz = int[].class;
+
+            // when
+            String result = ClassUtil.processClassByType(clazz, typeCallback);
+
+            // then
+            assertThat(result, equalTo("arrayType[" + int[].class.getName() + "]"));
+        }
+
+        @Test
+        void shouldUseInterfaceCallback() {
+            // given
+            ClassTypeCallback<String> typeCallback = new CallbackTestImpl();
+            Class<?> clazz = Runnable.class;
+
+            // when
+            String result = ClassUtil.processClassByType(clazz, typeCallback);
+
+            // then
+            assertThat(result, equalTo("interface[" + Runnable.class.getName() + "]"));
+        }
+
+        @Test
+        void shouldUseProxyClassCallback() {
+            // given
+            ClassTypeCallback<String> typeCallback = new CallbackTestImpl();
+            Class<?> clazz = getClass().getAnnotation(Nested.class).getClass();
+
+            // when
+            String result = ClassUtil.processClassByType(clazz, typeCallback);
+
+            // then
+            assertThat(result, equalTo("proxyClass[" + clazz.getName() + "]"));
+        }
+
+        @Test
+        void shouldUseRegularClassCallback() {
+            // given
+            ClassTypeCallback<String> typeCallback = new CallbackTestImpl();
```

Comment:
* Check that ClassTypeCallback returning `null` is fine
* Ensure in the tests that if we add a new enum entry to the class type, the `processClassByType` method is kept in sync